### ISSUE TITLE
Examples not shown with IntoResponses

### ISSUE THREAD

**Issue by hpsjakob** [juhaku/utoipa#779]
Hi,

I'm using utoipa as follows:

```rust
#[derive(IntoResponses)]
enum StandardResponses {
    #[response(status = 500, 
        content_type= "text/plain", 
        description = "Internal Error.", 
        example = json!("Internal error."))]
    InternalError,
}

#[utoipa::path(
    post,
    path = "/my_router",
    responses(StandardResponses),
    )]
pub async fn pair_device(){
  todo!()
}
```

This generates the following JSON output:
![grafik](https://github.com/juhaku/utoipa/assets/112160602/c10b56c7-9e91-4535-843a-b33277e408f9)

I'm missing the content section here. I would have expected it to be shown since I specified `content_type` and `example`.
Did I get somethign wrong or could it be that I found a bug?

I'm usign version 4.0.0.

Thank you in advance for your time!

**Comment by hpsjakob** [comment]
I found that this is not the case if I change the enum to
```rust
enum StandardResponses {
    #[response(status = 500, 
        content_type= "text/plain", 
        description = "Internal Error.", 
        example = json!("Internal error."))]
    InternalError(String),
}
```

So I guess utoipa assumes if it's an empty enum that there's now content.

**Comment by juhaku** [comment]
@hpsjakob Yeah this is related to #965. Currently I believe as you stated that if there is no content then the content-type will not be set. This is something that needs some tuning and we should support this eventually.

**Comment by juhaku** [comment]
Yup this is a bug and finally fixed here #1073
